kong/ids.go

```diff
@@ -128,6 +128,49 @@ func (v *Vault) FillID(workspace string) error {
 	return nil
 }
 
+// FillID fills the ID of an entity. It is a no-op if the entity already has an ID.
+// ID is generated in a deterministic way using UUIDv5.
+// The UUIDv5 namespace being used for generation is separate from other namespaces used for generating IDs for other types.
+// The name used to generate the ID for plugins is the combination of:
+// plugin.Name, plugin.InstanceName, name of plugin.Service, plugin.Route, plugin.Consumer, and plugin.ConsumerGroup.
+func (p *Plugin) FillID(workspace string) error {
+	if p == nil {
+		return fmt.Errorf("plugin is nil")
+	}
+	if p.ID != nil && len(*p.ID) > 0 {
+		// ID already set, do nothing.
+		return nil
+	}
+	if p.Name == nil || len(*p.Name) == 0 {
+		return fmt.Errorf("plugin name is required")
+	}
+
+	toHash := *p.Name
+	if p.InstanceName != nil && len(*p.InstanceName) > 0 {
+		toHash = toHash + ":instance_name/" + *p.InstanceName
+	}
+	if p.Service != nil {
+		toHash = toHash + ":service/" + p.Service.FriendlyName()
+	}
+	if p.Route != nil {
+		toHash = toHash + ":route/" + p.Route.FriendlyName()
+	}
+	if p.Consumer != nil {
+		toHash = toHash + ":consumer/" + p.Consumer.FriendlyName()
+	}
+	if p.ConsumerGroup != nil {
+		toHash = toHash + ":consumer_group/" + p.ConsumerGroup.FriendlyName()
+	}
```

Comment:
Nit: to make this ever so slightly more efficient we could use `strings.Builder` to reduce the number of string copies: https://pkg.go.dev/strings#Builder. WDYT?

Reply:
Done.